tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -447,7 +438,7 @@ def options(self):
     return self._options_attr
 
   def _apply_debug_options(self):
-    if DEBUG_MODE:
+    if structured_function.DEBUG_MODE:
```

Comment:
`DEBUG_MODE` should go into common.py (or stay in dataset_ops.py for now) since it is used for many things that aren't related to structured functions.

Reply:
@aaudiber thanks. I can put `DEBUG_MODE` in dataset_ops.py for now but then it would generate a circular dependency with `structured_function.py`. As you said, I can keep it in a separate `common.py` file but what else can we put in that file?

```
class Dataset(V1|V2) -> dataset_ops
class (F)Dataset -> respective f_dataset.py file
```
the remaining helper classes can't be moved as they will introduce a circular dependency.

Reply:
There is nothing wrong with resolving circular dependencies among files in `tensorflow/python/data/ops/` by using `lazy_loader.LazyLoader`. I think almost all of the files we create with this refactoring will need to lazy load `dataset_ops.py`.

Keeping `DEBUG_MODE` in `dataset_ops.py` sgtm.

Reply:
@aaudiber I see. Just wanted to keep laze loading as a last resort so wanted to discuss the decision with you. I have made the. necessary changes. Please take a look.